src/editor/Cloud.tsx

```diff
@@ -0,0 +1,66 @@
+import toast from "react-hot-toast"
+
+import { UploadFile } from "~/lib/upload-file"
+
+import { ICommand } from "."
+
+const uploadResources = async (text: string) => {
+  const markdownHttpImagesRegex = /\!\[([^\]]*)\]\((?=(https:\/\/))([^)]+)\)/g
+  const markdownHttpImageRegex = /\!\[([^\]]*)\]\((?=(https:\/\/))([^)]+)\)/
+  const markdownHttpImages = text.match(markdownHttpImagesRegex)
+  if (markdownHttpImages) {
+    const markdownReplacementsArray = await Promise.all(
+      markdownHttpImages.map(async (markdownHttpImage) => {
+        const match = markdownHttpImage.match(markdownHttpImageRegex)!
+        const altText = match[1]
+        const url = match[3]
+        const toastId = toast.loading(`Uploading ${altText || url}`)
+        try {
+          const blob = await fetch(url).then((response) => response.blob())
+          const key = (await UploadFile(blob)).key
+          toast.success(`Uploaded ${altText || url}!`, {
+            id: toastId,
+          })
+          return {
+            originalMarkdown: markdownHttpImage,
+            newMarkdown: `![${altText}](${key})`,
+          }
+        } catch (error) {
+          toast.error(`Failed upload ${altText || url}, ${error.message}`, {
```

Comment:
There will be type errors here, see CI check, we can modify it like this, put `error` in it

```ts
if (error instanceof Error) {

}
```